src/js/utils/hooks.ts

```diff
@@ -42,3 +42,31 @@ export const useDidUpdateEffect = (onUpdate, deps) => {
         firstRef.current = true;
     }, deps);
 };
+
+function updateUrlSearchParams(value, key) {
+    const params = new URLSearchParams(window.location.search);
+    params.set(key, value);
+    window.history.replaceState({}, "", `${window.location.pathname}?${params.toString()}`);
+}
+
+/**
+ * Hook for managing and synchronizing URL search parameters with a component's state
+ *
+ * @param key - The search parameter key to be managed
+ * @param defaultValue - The default value to use when the search parameter key is not present in the URL
+ * @returns Object - An object containing the current value and a function to set the URL search parameter
+ */
+export function useUrlSearchParams({ key, defaultValue }) {
```

Comment:
Nitpick: I would use positional arguments here. Using an object is more cumbersome and doesn't obviously provide a benefit here